### PR TITLE
CIRC-1183] Log errors related to immediate patron notices to Circ Log

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/api/support/APITests.java [hunk 1]
```diff
@@ -82,7 +82,7 @@
 
   private final boolean initialiseCirculationRules;
 
-  private final ResourceClient servicePointsClient = ResourceClient.forServicePoints();
+  protected final ResourceClient servicePointsClient = ResourceClient.forServicePoints();
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/support/APITests.java:85]: Revert

File: src/test/java/api/loans/DueDateScheduledNoticesTests.java [hunk 2]
```diff
@@ -41,6 +41,7 @@
 import api.support.http.IndividualResource;
 import api.support.http.ItemResource;
 import api.support.http.OkapiHeaders;
+import api.support.utl.PatronNoticeTestHelper;
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/DueDateScheduledNoticesTests.java:44]: Remove?

File: src/main/java/org/folio/circulation/domain/notice/combiner/LoanNoticeContextCombiner.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  @Override
+  public JsonObject buildCombinedNoticeContext(Collection<PatronNoticeEvent> events) {
+    return events.stream()
+      .map(PatronNoticeEvent::getNoticeContext)
+      .filter(Objects::nonNull)
+      .collect(collectingAndThen(
+        Collectors.toList(),
+        loans -> new JsonObject()
+          .put("user", createUserContext(events.iterator().next().getUser()))
+          .put("loans", loans)
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/notice/combiner/LoanNoticeContextCombiner.java:26]:
```suggestion
          .put("loans", contexts)
```

File: src/main/java/org/folio/circulation/resources/LoanNoticeSender.java [hunk 4]
```diff
@@ -57,23 +60,40 @@ private void sendLoanNotice(LoanAndRelatedRecords records, NoticeEventType event
     sendLoanNotice(records.getLoan(), eventType);
   }
 
-  private void sendLoanNotice(Loan loan, NoticeEventType eventType) {
+  private CompletableFuture<Result<Void>> sendLoanNotice(Loan loan, NoticeEventType eventType) {
+    return validateLoan(loan)
+      .after(l -> sendNotice(loan, eventType));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/LoanNoticeSender.java:65]: What happens when validation fails?

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**